### PR TITLE
Migrate spring data cosmos modules to sdk/spring dir

### PR DESCRIPTION
**Pull request by moarychan**
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/30793 https://github.com/Azure/azure-sdk-for-java/issues/30458

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by moarychan** [comment]
@kushagraThapar there are some test failures in the spring data auditable feature when enabling azure-spring-data-cosmos  integration tests, the attribute value of the date type returned by the cosmos service is larger than the actual value, and the difference is a nanosecond deviation, this doesn't happen every time. Could you help follow up on this issue in the main branch first?

**Comment by moarychan** [comment]
The below failures will be fixed later:
- Some broken links in branch 6.x, maybe a pipeline bug.
- Cosmos IT failures, which may be fixed in main branch first.
- Ignore the compatibility test for Spring Boot 3.0.0-M3, because the break changes in Spring Data from 3.0.0-M4 to 3.0.0-M5.
This PR is ready to merge, @saragluna please help to review, thanks!

**Review by jsquire — COMMENTED** [review]
Changes to CODEOWNERS and Fabric Bot rules look good to me.

File: eng/versioning/version_client.txt
Comment (kushagraThapar) [inline eng/versioning/version_client.txt:146]: current versions of azure-spring-data-cosmos is v3. Why do we have the version as 6.0.0-beta.1 ?
What is the plan for releasing multiple major versions of azure-spring-data-cosmos?

File: sdk/cosmos/tests.yml
Comment (kushagraThapar) [inline sdk/cosmos/tests.yml:39]: Why are we removing this? Can't we still run spring-data-cosmos-test integration pipelines on `azure-cosmos` module development?

File: sdk/spring/azure-spring-data-cosmos/pom.xml
Comment (kushagraThapar) [inline sdk/spring/azure-spring-data-cosmos/pom.xml:96]: We should still depend on current version of azure-cosmos like we do right now, so that we can continuously run the integration tests when we develop azure-cosmos module. 
If we don't do this, we won't be able to catch any regressions because of development on azure-cosmos module, which can then only be caught after the azure-cosmos module has been released.

File: sdk/spring/azure-spring-data-cosmos/pom.xml
Comment (kushagraThapar) [inline sdk/spring/azure-spring-data-cosmos/pom.xml:89]: why do we need micrometer in spring-data-cosmos?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
@moarychan great work. I have added few comments which are worth discussing before we merge this PR. 
Regarding your comment on `Cosmos IT failures, which may be fixed in main branch first.` 
There have been no failures regarding CosmosAuditing feature in our CI, and I think the code changes introduced in this PR related to auditing feature are causing the pipeline to fail. Please fix the tests before merging this PR.

File: sdk/spring/azure-spring-data-cosmos/pom.xml
Comment (moarychan) [reply sdk/spring/azure-spring-data-cosmos/pom.xml:89]: It's used to fix the issue  https://github.com/micrometer-metrics/micrometer/issues/3398

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md:126]: Shouldn't we stick with the Main branch here? My assumption was the feature branch is used for future versions - while Main contains the current stable version?
https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-data-cosmos/

File: sdk/cosmos/azure-cosmos-spark_3-2_2-12/README.md
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-2_2-12/README.md:125]: See above - stick with Main branch

File: eng/versioning/version_client.txt
Comment (FabianMeiswinkel) [inline eng/versioning/version_client.txt:146]: Why make this look like a breaking change (with major verson bump)?

File: sdk/cosmos/tests.yml
Comment (FabianMeiswinkel) [reply sdk/cosmos/tests.yml:39]: +1 - same comment for the ci.yml above. We would still need to compile and run the spring-data-cosmos tests for any changes in azure-cosmos

File: sdk/spring/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/PageableMemoRepository.java
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/PageableMemoRepository.java:14]: See above -please revert

File: sdk/spring/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/PageablePersonRepository.java
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/PageablePersonRepository.java:11]: see above - pelase revert

File: sdk/spring/azure-spring-data-cosmos/README.md
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos/README.md:621]: please stick to main

File: sdk/spring/azure-spring-data-cosmos/pom.xml
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos/pom.xml:15]: Not clear why this would be marked as breaking change (major version upgrade)? We probably need to discuss in a meeting - anytime we introduce a new major version we need to have a plan to deprecate/retire the previous one - and understand that we duplicate support cost.

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/CosmosRepository.java
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/CosmosRepository.java:19]: Please revert - not necessary PagingAndSortedRepository already extends CrudRepository - so CosmosRepository is implicitly extending CrudRepository already

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/ReactiveCosmosRepository.java
Comment (FabianMeiswinkel) [inline sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/ReactiveCosmosRepository.java:15]: Dito

File: sdk/spring/integration-matrix.json
Comment (FabianMeiswinkel) [inline sdk/spring/integration-matrix.json:9]: I guess adding Java 17 test coverage is goodness - but removing Java 11 test coverage is bad. Vast majority of our customers still uses Java 8 or 11

File: sdk/spring/azure-spring-data-cosmos/pom.xml
Comment (FabianMeiswinkel) [reply sdk/spring/azure-spring-data-cosmos/pom.xml:89]: But shouldn't it be sufficient to add this as "provided" vs. making it a transitive dependency?

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
Few comments  - also would like to understand the game plan on how to execute the move from sdk/cosmos to sdk/spring in main branch. My preference would have been to do the move in main first - to minimize the time where possibly new dev work / feature work is done in the feature branch and code diverges between master and feature branch more and more.

File: sdk/spring/integration-matrix.json
Comment (kushagraThapar) [reply sdk/spring/integration-matrix.json:9]: +1 to Fabian. Also, Java 17 doesn't work with spring-data-cosmos because of UUID support as Id type. There is an open bug https://github.com/Azure/azure-sdk-for-java/issues/28926

**Comment by moarychan** [comment]
Close this one, new PR is comming.

**Pull request closed without merging**
